### ISSUE TITLE
Generalization results of litemono on make3d

### ISSUE THREAD

**Issue by bizheyandebing** [noahzn/Lite-Mono#73]
This is a very valuable job. I have some confusion, I can reproduce your results in Table 2 based on the model you provided, 

![image](https://github.com/noahzn/Lite-Mono/assets/29300743/96537c69-df22-4dab-a31f-f0cc9bf47ee3)

but the results are different from Table 3 when I use the same model for make3d generalization testing.
![image](https://github.com/noahzn/Lite-Mono/assets/29300743/2120e1f3-7930-4b8d-a295-08b6fa79b5fe)

According to the description in the article, they should be the same model. Is there something wrong with my evaluation code?

**Comment by bizheyandebing** [comment]
`from __future__ import absolute_import, division, print_function
from scipy import io
import os
import cv2
import numpy as np
import torch
from torch.utils.data import DataLoader
from layers import disp_to_depth
from utils import readlines
from options import LiteMonoOptions
import datasets
import networks
import time

options = LiteMonoOptions()
opt=options.parse()
load_weights_folder = '/home/ubuntu/tmp/lite_y/lite-mono_640x192'
main_path = "/home//MAKE3D"
encoder_path = os.path.join(load_weights_folder, "encoder.pth")
decoder_path = os.path.join(load_weights_folder, "depth.pth")
encoder_dict = torch.load(encoder_path)
decoder_dict = torch.load(decoder_path)
encoder = networks.LiteMono(model=opt.model,
                            height=encoder_dict['height'],
                            width=encoder_dict['width'])
depth_decoder = networks.DepthDecoder(encoder.num_ch_enc, scales=range(3))
model_dict = encoder.state_dict()
depth_model_dict = depth_decoder.state_dict()
encoder.load_state_dict({k: v for k, v in encoder_dict.items() if k in model_dict})
depth_decoder.load_state_dict({k: v for k, v in decoder_dict.items() if k in depth_model_dict})

encoder.cuda()
encoder.eval()
depth_decoder.cuda()
depth_decoder.eval()


def compute_errors(gt, pred):
    rmse = (gt - pred) ** 2
    rmse = np.sqrt(rmse.mean())

    rmse_log = (np.log10(gt) - np.log10(pred)) ** 2
    rmse_log = np.sqrt(rmse_log.mean())

    abs_rel = np.mean(np.abs(gt - pred) / gt)

    sq_rel = np.mean(((gt - pred) ** 2) / gt)

    return abs_rel, sq_rel, rmse, rmse_log

def save_ima(image,daph,i):
    #target_depth0=image.cpu().detach().numpy()
    # print(target_depth0.shape)
    image=np.squeeze(image)
    #pdb.set_trace()
   # print(target_depth0.shape)
    #disp_resized_np =image.transpose(1,2,0)
    #disp_resized_np=disp_resized_np[:,:,(2,1,0)]
    path = os.path.join("/home/ubuntu/ubuntu_jixie/", daph,)
    #name_dest_im = os.path.join("/home/wyy/tmp/gt-depth/", "{}_gt_disp.jpeg".format(i))
    if not os.path.exists(path):
        os.mkdir(path)
    loder = os.path.join("/home/ubuntu/ubuntu_jixie/", daph, "{}.jpeg".format(i))
    cv2.imwrite(loder,image)
with open(os.path.join(main_path, "make3d_test_files.txt")) as f:
    test_filenames = f.read().splitlines()
test_filenames = map(lambda x: x[4:-4], test_filenames)

depths_gt = []
images = []
ratio = 2
h_ratio = 1 / (1.33333 * ratio)
color_new_height = int(1704 / 2)
depth_new_height = 21
for filename in test_filenames:
    mat = io.loadmat(os.path.join(main_path, "Gridlaserdata", "depth_sph_corr-{}.mat".format(filename)))
    depths_gt.append(mat["Position3DGrid"][:, :, 3])
    #print(filename)
    image = cv2.imread(os.path.join(main_path, "Test134", "img-{}.jpg".format(filename)))
    #print(filename)
    image = image[int((2272 - color_new_height) / 2):int((2272 + color_new_height) / 2), :, :]

    images.append(image[:, :, ::-1])
depths_gt_resized = map(lambda x: cv2.resize(x, (305, 407), interpolation=cv2.INTER_NEAREST), depths_gt)
depths_gt_cropped = map(lambda x: x[int((55 - 21) / 2):int((55 + 21) / 2), :], depths_gt)

depths_gt_cropped = list(depths_gt_cropped)
errors = []
with torch.no_grad():
    for i in range(len(images)):
        input_color = images[i]
        #save_ima(input_color[:,:,::-1], "make3d_y", i)
        input_color = cv2.resize(input_color / 255.0, (512, 256), interpolation=cv2.INTER_NEAREST)

        input_color = torch.tensor(input_color, dtype=torch.float).cuda().permute(2, 0, 1)[None, :, :, :]

        output = depth_decoder(encoder(input_color))

        depth_gt = depths_gt_cropped[i]
        _,depth_pred = disp_to_depth(output[("disp", 0)], 1e-3,70)

        # disp_resized_np = output[("disp", 0)].cpu().squeeze()
        # vmax = np.percentile(disp_resized_np, 95)
        # normalizer = mpl.colors.Normalize(vmin=disp_resized_np.min(), vmax=vmax)
        # # normalizer = mpl.colors.Normalize(vmin=vmin, vmax=np.percentile(disp_resized_np, 95))
        # mapper = cm.ScalarMappable(norm=normalizer, cmap='magma')
        # colormapped_im = (mapper.to_rgba(disp_resized_np)[:, :, :3] * 255).astype(np.uint8)
        # im = pil.fromarray(colormapped_im)
        # name_dest_im = os.path.join(load_weights_folder, "Make3D","{}_pred_disp.jpeg".format(i))
        # # name_dest_im = os.path.join("/home/wyy/tmp/gt-depth/", "{}_gt_disp.jpeg".format(i))
        # im.save(name_dest_im)


        depth_pred =depth_pred.squeeze().cpu().numpy()
        depth_pred = cv2.resize(depth_pred, depth_gt.shape[::-1], interpolation=cv2.INTER_NEAREST)
        mask = np.logical_and(depth_gt > 0, depth_gt < 70)
        depth_gt = depth_gt[mask]
        depth_pred = depth_pred[mask]
        depth_pred *= np.median(depth_gt) / np.median(depth_pred)
        depth_pred[depth_pred > 70] = 70
        errors.append(compute_errors(depth_gt, depth_pred))
    mean_errors = np.mean(errors, 0)

print(("{:>8} | " * 4).format("abs_rel", "sq_rel", "rmse", "rmse_log"))
print(("{: 8.3f} , " * 4).format(*mean_errors.tolist()))`

**Comment by bizheyandebing** [comment]
I think I solved this problem, you answered me before in monodepth2's discussion forum, but I forgot. Thank you very much, your work is very valuable.


https://github.com/nianticlabs/monodepth2/issues/392#issuecomment-1128977268
